src/main/java/duke/command/FindCommand.java

```diff
@@ -0,0 +1,68 @@
+package duke.command;
+
+import java.util.ArrayList;
+
+import duke.exception.DukeException;
+import duke.storage.Storage;
+import duke.task.DukeTask;
+import duke.tasklist.TaskList;
+import duke.ui.Ui;
+
+/**
+ * Encapsulates searching for keywords in a TaskList.
+ *
+ * @author Kartikeya
+ */
+public class FindCommand implements Command {
+    private final String[] input;
+
+    /**
+     * Constructor for a FindCommand object.
+     *
+     * @param input split strings input by user
+     */
+    public FindCommand(String[] input) {
+        this.input = input;
+    }
+
+    /**
+     * {@inheritDoc}
+     * Finds tasks matching input string, and shows output to user.
+     */
+    @Override
+    public void execute(TaskList itemList, Ui ui, Storage storage) throws DukeException {
+        ui.showToUser(executeFindItems(input, itemList));
+    }
+
+    /**
+     * Parses input array of strings to a string that can be used to call the find method
+     * on itemList. Calls the method and parses output to a format suitable for the ui,
+     * then returns it. Accepts an empty input after the find keyword, and returns all the
+     * tasks stored in this case. Throws a DukeException if no items were found.
```

Comment:
Perhaps instead of explaining how it works, you could focus on the What (What is this method/code supposed to do?) or Why (Rationale for the implementation)?

Reply:
Thank you, I've changed the Javadoc comment for this!